test/MOI_wrapper.jl

```diff
@@ -0,0 +1,69 @@
+using Test
+using LinearAlgebra
+using Manifolds
+using Manopt
+using JuMP
+
+function test_sphere()
+    model = Model(Manopt.Optimizer)
+    start = normalize(1:3)
+    @variable(model, x[i=1:3] in Sphere(2), start = start[i])
+
+    @objective(model, Min, sum(x))
+    optimize!(model)
+    @test termination_status(model) == MOI.LOCALLY_SOLVED
+    @test primal_status(model) == MOI.FEASIBLE_POINT
+    @test primal_status(model) == MOI.FEASIBLE_POINT
+    @test dual_status(model) == MOI.NO_SOLUTION
+    @test value.(x) ≈ -inv(√3) * ones(3) rtol = 1e-2
+    @test raw_status(model) isa String
+    @test raw_status(model)[end] != '\n'
+
+    @objective(model, Max, sum(x))
+    set_start_value.(x, start)
+    optimize!(model)
+    @test value.(x) ≈ inv(√3) * ones(3) rtol = 1e-2
+    @test raw_status(model) isa String
+    @test raw_status(model)[end] != '\n'
+
+    # Creating a model directly with `@NLobjective` wouldn't work
+    # because of https://github.com/jump-dev/MathOptInterface.jl/blob/32dbf6056b0b5fb9d44dc583ecc8249f6fd703ea/src/Utilities/copy.jl#L485-L500
+    # so we need to wait for https://github.com/jump-dev/JuMP.jl/pull/3106 for NL objectives to work with `@objective`
+    # Here, we are by-passing that because we are modifying the objective
+    # so we don't call `copy_to`.
```

Comment:
Why wouldn't this work?

Reply:
When there is an `NLPBlock`, the variables are not added as constrained variables. Here we only want to support constrained variables.
This is going to be fixed by https://github.com/jump-dev/JuMP.jl/pull/3106 anyway

Reply:
If we really want to make it work we could add a method for `add_variable` and `add_constraint`. That's not going to send the right info to the bridges if we say we support free variables and this is the sources of issues like https://github.com/jump-dev/MosekTools.jl/issues/115 so I'd rather not do it.

Reply:
This one is still open – what do we do about it? A next PR?

Reply:
https://github.com/jump-dev/JuMP.jl/pull/3106 has been merged and release and we use this new JuMP interface (which is why I waited to update this PR) so now there is no issue anymore